pkgs/pkgs-lib/formats.nix

```diff
@@ -70,16 +70,23 @@ rec {
       };
 
       iniAtom =
-        if listsAsDuplicateKeys then
+        if listsAsDuplicateKeys || listToValue != null then
           coercedTo singleIniAtom lib.singleton (listOf singleIniAtom) // {
-            description = singleIniAtom.description + " or a list of them for duplicate keys";
+            description = singleIniAtom.description + " or a list of them";
           }
         else
           singleIniAtom;
 
     in attrsOf (attrsOf iniAtom);
 
-    generate = name: value: pkgs.writeText name (lib.generators.toINI args value);
+    generate = name: value:
+      let
```

Comment:
```suggestion
      assert !listsAsDuplicateKeys || listToValue == null;
      let
```

Reply:
Moved the assertion to the format definition start